tests/host_tools/cargo_build.py

```diff
@@ -49,35 +51,30 @@ def cargo_test(path, extra_args=""):
     utils.run_cmd(cmd)
 
 
+@with_filelock
 def get_firecracker_binaries():
     """Build the Firecracker and Jailer binaries if they don't exist.
 
     Returns the location of the firecracker related binaries eventually after
     building them in case they do not exist at the specified root_path.
     """
     target = DEFAULT_BUILD_TARGET
-    out_dir = "{target_dir}/{target}/release".format(
-        target_dir=FC_WORKSPACE_TARGET_DIR, target=target
-    )
-    fc_bin_path = "{}/{}".format(out_dir, FC_BINARY_NAME)
-    jailer_bin_path = "{}/{}".format(out_dir, JAILER_BINARY_NAME)
-
-    if getattr(get_firecracker_binaries, "binaries_built", False):
-        return fc_bin_path, jailer_bin_path
-
-    cd_cmd = "cd {}".format(FC_WORKSPACE_DIR)
-    flags = 'RUSTFLAGS="{}"'.format(get_rustflags())
-    cargo_default_cmd = "cargo build --release --target {}".format(target)
-    cargo_jailer_cmd = "cargo build -p jailer --release --target {}".format(target)
-    cmd = "{0} && {1} {2} && {1} {3}".format(
-        cd_cmd, flags, cargo_default_cmd, cargo_jailer_cmd
-    )
-
-    utils.run_cmd(cmd)
-
-    utils.run_cmd("strip --strip-debug {} {}".format(fc_bin_path, jailer_bin_path))
+    target_dir = FC_WORKSPACE_TARGET_DIR
+    out_dir = Path(f"{target_dir}/{target}/release")
+    fc_bin_path = out_dir / FC_BINARY_NAME
+    jailer_bin_path = out_dir / JAILER_BINARY_NAME
+
+    if not fc_bin_path.exists():
+        cd_cmd = "cd {}".format(FC_WORKSPACE_DIR)
+        flags = 'RUSTFLAGS="{}"'.format(get_rustflags())
+        cargo_default_cmd = f"cargo build --release --target {target}"
+        cargo_jailer_cmd = f"cargo build -p jailer --release --target {target}"
```

Comment:
Why can't we use here the `cargo_build` function from above?

Reply:
I didn't try. Looking at it it's not immediately obvious as `cargo_build` uses `CARGO_TARGET_DIR` so that could have some side effects. I agree it would be good to clean that up, but I consider it out of scope for this change.